src/arrayMethodJoin.js

```diff
@@ -4,8 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (!this) {
```

Comment:
Навіть пустий масив конвертується в true, тому це if ніколи не виконається
```suggestion
    if (!this.length) {
```